### PR TITLE
Added ticket release date, ticket available and action link to event

### PR DESCRIPTION
**Pull request by AndrewBestbier**
Winning

Bonjour!

Add a ticketReleaseDate and a boolean: ticketAvailability, for the specified event. This gives all the required information to move the react.london client to different event states automatically.

### CODE REVIEW

File: lib/event/eventFields.js [hunk 1]
```diff
@@ -73,6 +74,18 @@ export const EventType = new GraphQLObjectType({
       type: DateTimeType,
       resolve: mapDateTime('endDateTime'),
     },
+    ticketReleaseDate: {
+      type: DateTimeType,
+      resolve: mapDateTime('ticketReleaseDate'),
+    },
+    ticketsAvailiable: {
+      type: GraphQLBoolean,
+      description: 'A boolean show whether tickets are availiable',
```
Comment (bkspace) [inline lib/event/eventFields.js]: 'A boolean *to* show...'

File: lib/event/eventFields.js [hunk 2]
```diff
@@ -73,6 +74,18 @@ export const EventType = new GraphQLObjectType({
...
     },
+    ticketReleaseDate: {
+      type: DateTimeType,
+      resolve: mapDateTime('ticketReleaseDate'),
+    },
+    ticketsAvailiable: {
+      type: GraphQLBoolean,
+      description: 'A boolean to show whether tickets are availiable',
+    },
+    eventActionLink: {
```
Comment (asavin) [inline lib/event/eventFields.js]: i feel like there's potentially some duplication going on with the `externalLinks` and `internalLinks`. One option I think would be to extend `RelatedLinkType` to be featured as a "main" event action link.

File: lib/event/eventFields.js [hunk 3]
```diff
@@ -73,6 +74,14 @@ export const EventType = new GraphQLObjectType({
       type: DateTimeType,
       resolve: mapDateTime('endDateTime'),
     },
+    ticketReleaseDate: {
+      type: DateTimeType,
+      resolve: mapDateTime('ticketReleaseDate'),
```
Comment (AndrewBestbier) [inline lib/event/eventFields.js]: Needs description

File: lib/event/eventFields.js [hunk 4]
```diff
@@ -73,6 +74,15 @@ export const EventType = new GraphQLObjectType({
       type: DateTimeType,
       resolve: mapDateTime('endDateTime'),
     },
+    ticketReleaseDate: {
+      type: DateTimeType,
+      description: 'The date when the event tickets are released.',
+      resolve: mapDateTime('ticketReleaseDate'),
+    },
+    ticketsAvailiable: {
```
Comment (asavin) [inline lib/event/eventFields.js]: typo

File: lib/event/eventFields.spec.js [hunk 5]
```diff
@@ -33,15 +33,29 @@ describe('Event Queries', () => {
...
+          externalLinks {
+            title
+            url
+            type
+          }
           schedule {
             datetime
             text
           }
+          ticketsAvailiable
```
Comment (asavin) [inline lib/event/eventFields.spec.js]: typo

File: lib/fetch.spec.js [hunk 6]
```diff
@@ -185,4 +188,28 @@ describe('sanitizeTalk', () => {
       ],
     });
   });
+
+  describe('RefromatLinks', () => {
```
Comment (asavin) [inline lib/fetch.spec.js]: typo

File: prismic/custom-types/events.json [hunk 7]
```diff
@@ -47,6 +47,20 @@
         "label" : "End date and time"
       }
     },
+    "ticketReleaseDate" : {
+      "type" : "Timestamp",
+      "config" : {
+        "label" : "Ticket release date"
+      }
+    },
+    "ticketsAvailiable" : {
```
Comment (asavin) [inline prismic/custom-types/events.json]: typo

File: prismic/custom-types/events.json [hunk 8]
```diff
@@ -47,6 +47,20 @@
...
+      "type" : "Select",
+      "fieldset" : "A boolean showing whether the tickets are availiable",
```
Comment (asavin) [inline prismic/custom-types/events.json]: typo

File: prismic/custom-types/events.json [hunk 9]
```diff
@@ -47,6 +47,20 @@
...
+      "config" : {
+        "label" : "Tickets Availiable",
```
Comment (asavin) [inline prismic/custom-types/events.json]: typo

**Pull request merged**